lib/ansible/modules/find.py

```diff
@@ -344,6 +366,26 @@ def contentfilter(fsname, pattern, read_whole_file=False):
     return False
 
 
+def mode_filter(st, mode, exact, module):
+    if not mode:
+        return True
+
+    st_mode = stat.S_IMODE(st.st_mode)
+
+    if not isinstance(mode, int):
```

Comment:
I still think we should not accept ints, using raw you can still implement 'strict strings' check on module side

Reply:
Done.